File: algorithm/45. Jump Game II/45.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ret,curmax,currch=0,0,0
        for i in xrange(len(nums)):
            if currch < i:
                ret+=1
                currch=curmax
            curmax=max(curmax, i+nums[i])
        return ret

File: algorithm/45. Jump Game II/45.java
class Solution {
    public int jump(int[] nums) {
        int res = 0;
        int dis_res = 0;
        int dis_max = 0;
        for(int i = 0; i < nums.length - 1; i++) {
            dis_max = Math.max(dis_max, i + nums[i]);
            if(dis_res == i) {
                res++;
                dis_res = dis_max;
            }
        }
        return res;
    }
}
